### ISSUE TITLE
consider-using-enumerate shown for loop where enumeration is impossible

### ISSUE THREAD

**Issue by tcobbs-bentley** [PyCQA/pylint#7942]
### Bug description

The consider-using-enumerate rule is falsely saying that I should use enumerate in a situation when doing so is impossible. The following is an example:

``` py
def pylint_bug():
    '''
    Demonstration of bug in pylint's consider-using-enumerate rule.
    '''

    max_lengths = [ 0, 0 ]
    rows = [
        [ "hello", "goodbye" ],
        [ "hola", "adios" ]
    ]
    for row in rows:
        for i in range(len(max_lengths)):
            max_lengths[i] = max(max_lengths[i], len(row[i]))
```

### Command used

```shell
Visual Studio Code's "Python: Select Linter".
```

### Pylint output

```shell
[{
	"resource": "<...>/pylint_bug.py",
	"owner": "python",
	"code": "consider-using-enumerate",
	"severity": 2,
	"message": "Consider using enumerate instead of iterating with range and len",
	"source": "pylint",
	"startLineNumber": 35,
	"startColumn": 9,
	"endLineNumber": 36,
	"endColumn": 62
}]
```

### Expected behavior

If a loop cannot be rewritten using an enumeration (such as when modifying the values being enumerated), the consider-using-enumerate rule should not be triggered.

### Pylint version

```shell
No idea how to determine that using pylint inside Visual Studio Code. I selected pylint as my linter today (2022-12-24), and pylint was installed by VS Code when I did that.
```

### OS / Environment

macOS Monterey 12.6
Visual Studio Code 1.74.1

**Comment by nickdrozd** [comment]
I suppose the expected rewrite would look like this:

```python
    for row in rows:
        for i, length in enumerate(max_lengths):
            max_lengths[i] = max(length, len(row[i]))
```

That works as far as I can tell. And my understanding is that modifying list values during iteration is fine so long as the list length doesn't change. Or am I missing something?

**Comment by Pierre-Sassoulas** [comment]
Thanks @nickdrozd, this is the supposed fix.
